src/material/slider/slider.ts

```diff
@@ -647,6 +589,86 @@ export class MatSlider extends _MatSliderMixinBase
     this._isSliding = false;
   }
 
+  /** Called when the user has put their pointer down on the slider. */
+  private _pointerDown = (event: TouchEvent | MouseEvent) => {
+    // Don't do anything if the slider is disabled or the
+    // user is using anything other than the main mouse button.
+    if (this.disabled || this._isSliding || (!isTouchEvent(event) && event.button !== 0)) {
+      return;
+    }
+
+    this._runInsideZone(() => {
+      const oldValue = this.value;
+      this._isSliding = true;
+      event.preventDefault();
+      this._focusHostElement();
+      this._onMouseenter(); // Simulate mouseenter in case this is a mobile device.
+      this._bindGlobalEvents(event);
+      this._focusHostElement();
+      this._updateValueFromPosition(getPointerPositionOnPage(event));
+      this._valueOnSlideStart = this.value;
+
+      // Emit a change and input event if the value changed.
+      if (oldValue != this.value) {
+        this._emitInputEvent();
+        this._emitChangeEvent();
+      }
+    });
+  }
+
+  /**
+   * Called when the user has moved their pointer after
+   * starting to drag. Bound on the document level.
+   */
+  private _pointerMove = (event: TouchEvent | MouseEvent) => {
+    if (this._isSliding) {
+      // Prevent the slide from selecting anything else.
+      event.preventDefault();
+      const oldValue = this.value;
+      this._updateValueFromPosition(getPointerPositionOnPage(event));
+
+      // Native range elements always emit `input` events when the value changed while sliding.
+      if (oldValue != this.value) {
+        this._emitInputEvent();
+      }
+    }
+  }
+
+  /** Called when the user has lifted their pointer. Bound on the document level. */
+  private _pointerUp = (event: TouchEvent | MouseEvent) => {
+    if (this._isSliding) {
+      event.preventDefault();
+      this._isSliding = false;
+      this._removeGlobalEvents();
+
+      if (this._valueOnSlideStart != this.value && !this.disabled) {
+        this._emitChangeEvent();
+      }
+      this._valueOnSlideStart = null;
+    }
+  }
+
+  /** Binds our global move and end events. */
+  private _bindGlobalEvents(triggerEvent: TouchEvent | MouseEvent) {
+    if (typeof document !== 'undefined' && document) {
```

Comment:
Can you add a comment that explains the intent of the global events any that they're only necessary while actively sliding? (I assume it's so that your pointer doesn't have to stay on the tiny little slider track while moving)